### PR TITLE
Feature/request speed change

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description

Add API::requestSpeedChange function.

## How to review this PR.

- Execute commands below and check test scenario works well.
```
ros2 launch cpp_mock_scenarios mock_test.launch.py scenario:=request_speed_change launch_rviz:=True
```
- Please check passing all test cases.

### CODE REVIEW

**Review by yamacir-kit — APPROVED** [review]

**Review by yamacir-kit — CHANGES_REQUESTED** [review]
Cancel approvement.

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp [hunk 1]
```diff
@@ -146,6 +147,10 @@ class EntityBase
 
   virtual void requestAssignRoute(const std::vector<geometry_msgs::msg::Pose> & waypoints) = 0;
 
+  virtual void requestSpeedChange(
+    const double target_speed, const SpeedChangeTransition transition,
+    const SpeedChangeConstraint constaint, const bool continuous);
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp:152]: `constaint` => `constraint`

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp [hunk 2]
```diff
@@ -246,6 +247,10 @@ class EntityManager
 
   void setTargetSpeed(const std::string & name, double target_speed, bool continuous);
 
+  void requestSpeedChange(
+    const std::string & name, const double target_speed, const SpeedChangeTransition transition,
+    const SpeedChangeConstraint constaint, const bool continuous);
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp:252]: `constaint` => `constraint`

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp [hunk 3]
```diff
@@ -107,6 +107,10 @@ class PedestrianEntity : public EntityBase
 
   void requestAssignRoute(const std::vector<geometry_msgs::msg::Pose> &) override;
 
+  void requestSpeedChange(
+    const double target_speed, const SpeedChangeTransition transition,
+    const SpeedChangeConstraint constaint, const bool continuous) override;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp:112]: `constaint` => `constraint`

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp (hunk 3 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/648/commits/b608d5e2c700eea9af558ed92a2a6d2699994108

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp (hunk 2 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/648/commits/b608d5e2c700eea9af558ed92a2a6d2699994108

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/648/commits/b608d5e2c700eea9af558ed92a2a6d2699994108

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**